openff/toolkit/topology/topology.py

```diff
@@ -602,7 +602,15 @@ def box_vectors(self, box_vectors):
             self._box_vectors = None
             return
         if not hasattr(box_vectors, "units"):
-            raise InvalidBoxVectorsError("Given unitless box vectors")
+
+            if hasattr(box_vectors, "unit"):
+                # this is probably an openmm.unit.Quantity; we should gracefully import OpenMM but
+                # the chances of this being an object with the two previous conditions met is low
+                box_vectors = ensure_quantity(box_vectors, "openff")
```

Comment:
This case could optionally be an error, but I think it's better to just convert it. Setters don't have guarantees around not modifying inputs and the exception would just instruct users to do this.